pynav/filters.py

```diff
@@ -365,6 +411,342 @@ def _get_cost_and_info(
+        dt: float = None,
+        input_covariance: np.ndarray = None,
+    ) -> StateWithCovariance:
+        """
+        Propagates the state forward in time using a process model. The user
+        must provide the current state, input, and time interval
+
+        .. note::
+            If the time interval `dt` is not provided in the arguments, it will
+            be taken as the difference between the input stamp and the state stamp.
+
+        Parameters
+        ----------
+        x : StateWithCovariance
+            The current state estimate.
+        u : Input
+            Input measurement to be given to process model
+        dt : float, optional
+            Duration to next time step. If not provided, dt will be calculated
+            with `dt = u.stamp - x.state.stamp`.
+        input_covariance: np.ndarray, optional
+            Covariance associated to the inpu measurement. If not provided,
+            it will be grabbed from u.covariance
+        Returns
+        -------
+        StateWithCovariance
+            New predicted state
+        """
+
+        # Make a copy so we dont modify the input
+        x = x.copy()
+
+        # If state has no time stamp, load from measurement.
+        # usually only happens on estimator start-up
+        if x.state.stamp is None:
+            x.state.stamp = u.stamp
+
+        if input_covariance is None:
+            if u.covariance is not None:
+
+                input_covariance = u.covariance
+            else:
+                raise ValueError(
+                    "Input covariance information must be provided."
+                )
+
+        if dt is None:
+            dt = u.stamp - x.state.stamp
+
+        if dt < 0:
+            raise RuntimeError("dt is negative!")
+
+        if u is not None:
+
+            n_x = x.state.dof
+            n_u = u.dof
+
+            P = np.block(
+                [
+                    [x.covariance, np.zeros((n_x, n_u))],
+                    [np.zeros((n_u, n_x)), input_covariance],
+                ]
+            )
+
+            P_sqrt = np.linalg.cholesky(P)
+
+            n = n_x + n_u
+
+            if (n, self.method) in self._sigmapoint_cache:
+                unit_sigmapoints, w = self._sigmapoint_cache[(n, self.method)]
+            else:
+                unit_sigmapoints, w = generate_sigmapoints(n, self.method)
+                self._sigmapoint_cache[(n, self.method)] = (unit_sigmapoints, w)
+
+            sigmapoints = P_sqrt @ unit_sigmapoints
+
+            n_sig = w.size
+            x_propagated = []
+
+            # Propagate
+            for i in range(n_sig):
+                x_propagated.append(
+                    self.process_model.evaluate(
+                        x.state.plus(sigmapoints[0:n_x, i]),
+                        u.plus(sigmapoints[n_x::, i]),
+                        dt,
+                    )
+                )
+
+            # Compute mean. This probably wont work for composite states
```

Comment:
```suggestion
            # Compute mean. 
```

I think it will now! :)